public/video-ui/src/components/FormFields/FormFieldBylinePicker.js

```diff
@@ -0,0 +1,241 @@
+import React from 'react';
+import ContentApi from '../../services/capi';
+import { tagsFromStringList, tagsToStringList } from '../../util/tagParsers';
+import removeStringTagDuplicates from '../../util/removeStringTagDuplicates';
+
+export default class FormFieldBylinePicker extends React.Component {
+  state = {
+    bylineTags: [],
+    inputString: '',
+    lastAction: 'OTHER',
+    tagValue: []
+  };
+
+  componentDidMount() {
+    if (this.props.fieldValue !== this.props.placeholder) {
+      tagsFromStringList(this.props.fieldValue).then(result => {
+        this.setState({
+          tagValue: result
+        });
+      });
+    }
+  }
+
+  onUpdate = newValue => {
+    this.setState({
+      tagValue: newValue
+    });
+    this.props.onUpdateField(tagsToStringList(newValue));
+  };
+
+  updateInput = e => {
+    if (this.state.lastAction === 'SPACE') {
+      const newFieldValue = this.state.tagValue.concat([
+        this.state.inputString
+      ]);
+      this.onUpdate(newFieldValue);
+      this.setState({
+        inputString: ''
+      });
+
+      // If the user did not add new text input, we update the tag search
+    } else if (this.state.lastAction === 'DELETE') {
+      const length = this.state.tagValue.length;
+      const lastInput = this.state.tagValue[length - 1];
+
+      this.setState({
+        inputString: lastInput,
+        lastAction: 'OTHER'
+      });
+
+      const newValue = this.state.tagValue.slice(
+        0,
+        this.state.tagValue.length - 1
+      );
```

Comment:
This is the pre-commit hook reorganising code. If it is difficult to read we should probably change how it works.